### PR TITLE
Add peformance test around benchmark instrument for l2

### PR DESCRIPTION
**Pull request by OwenArnold**
Description of work.

Add a performance test which describes the fastest mechanism for calculating `L2` via the `MatrixWorkspace` interface. This is a useful comparison for an identical instrument setup used in the current [DMSC Instrument Prototype](https://github.com/DMSC-Instrument-Data/instrument-prototype). 

Check that the newly added performance test is registered and working. Other than that, code review.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/API/test/MatrixWorkspaceTest.h [hunk 1]
```diff
@@ -1402,8 +1433,26 @@ class MatrixWorkspaceTestPerformance : public CxxTest::TestSuite {
...
+    /*
+     * Simulate the L2 calculation performed via the Workspace/Instrument
+     * interface.
+     */
+    auto instrument = m_sansTestWorkspace.getInstrument();
+    auto sample = instrument->getSample();
+    double l2 = 0;
+    for (size_t i = 0; i < m_sansTestWorkspace.getNumberHistograms(); ++i) {
+      auto detector = m_sansTestWorkspace.getDetector(i);
+      l2 += detector->getDistance(*sample);
```
Comment (SimonHeybrock) [inline Framework/API/test/MatrixWorkspaceTest.h]: @OwenArnold If I remember correctly there is a caching mechanism for the positions of parent components. Do you know if the cache would be used in the case of this instrument? If so, the benchmark here is basically also measuring the time to initialize the cache. Have you tried adding a second test that does exactly the same thing again, to see if the performance is significantly different?

Comment (OwenArnold) [inline Framework/API/test/MatrixWorkspaceTest.h]: @SimonHeybrock. No I haven't, but that would be a good thing to add in addition to what we have.

**Comment by OwenArnold** [comment]
Across the two implementations. L2 access occurs with frequency for the whole instrument of.

- 2Hz for `Instrument 1.0` in Mantid
- 2000Hz for `Instrument 2.0` in Mantid

**Comment by SimonHeybrock** [comment]
Looks good. :shipit:

**Comment by FedeMPouzols** [comment]
This branch has developed a merge conflict.

**Comment by SimonHeybrock** [comment]
Owen is back on Monday, but there is not reason that would require having this in 3.8, I'll move it to 3.9.

**Comment by OwenArnold** [comment]
@SimonHeybrock @FedeMPouzols Thanks for flagging this up. This feature shouldn't hold up the release, but I'll try to resolve it asap anyway.

**Pull request merged**